snuba/query/snql/parser.py

```diff
@@ -35,17 +72,28 @@
     visit_parameters_list,
     visit_quoted_literal,
 )
+from snuba.util import parse_datetime
+
 
 snql_grammar = Grammar(
     r"""
-    query_exp             = match_clause where_clause? collect_clause? group_by_clause? having_clause? order_by_clause?
+    query_exp             = match_clause select_clause group_by_clause? where_clause? having_clause? order_by_clause? limit_by_clause? limit_clause? offset_clause? sample_clause?
 
-    match_clause          = space* "MATCH" space* open_paren clause close_paren space*
-    where_clause          = space* "WHERE" or_expression space*
-    collect_clause        = space* "COLLECT" collect_list space*
+    match_clause          = space* "MATCH" space* (relationship_match+ / entity_match / subquery) space*
+    select_clause         = space* "SELECT" select_list space*
     group_by_clause       = space* "BY" group_list space*
+    where_clause          = space* "WHERE" or_expression space*
     having_clause         = space* "HAVING" or_expression space*
     order_by_clause       = space* "ORDER BY" order_list space*
+    limit_by_clause       = space* "LIMIT" space* integer_literal space* "BY" space* column_name space*
+    limit_clause          = space* "LIMIT" space* integer_literal space*
+    offset_clause         = space* "OFFSET" space* integer_literal space*
+    sample_clause         = space* "SAMPLE" space* numeric_literal space*
+
+    entity_match          = open_paren entity_alias colon space* entity_name sample_clause? close_paren
+    relationship_link     = ~r"-\[" relationship_name ~r"\]->"
+    relationship_match    = space* entity_match space* relationship_link space* entity_match space* comma*
```

Comment:
Wouldn't the match clause match both
```
MATCH (e: Event)-[something]->(g: Groups)(e: Event)-[something]->(g: Assignees)
and
MATCH (e: Event)-[something]->(g: Groups), (e: Event)-[something]->(g: Assignees)
```
(since the comma has a star)
Is this intended?